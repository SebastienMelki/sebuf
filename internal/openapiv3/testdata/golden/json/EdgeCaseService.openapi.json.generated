{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"HeaderRequest":{"description":"Simple request message","properties":{"data":{"description":"Request data","type":"string"}},"type":"object"},"HeaderResponse":{"description":"Simple response message","properties":{"result":{"description":"Response data","type":"string"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"EdgeCaseService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/edge-headers/complex":{"post":{"description":"Method with complex header combinations","operationId":"ComplexHeaders","parameters":[{"description":"Array header with complex format","in":"header","name":"X-Complex-Array","required":true,"schema":{"type":"array"}},{"description":"Edge case header","in":"header","name":"X-Edge-Case","required":false,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ComplexHeaders","tags":["EdgeCaseService"]}}}}