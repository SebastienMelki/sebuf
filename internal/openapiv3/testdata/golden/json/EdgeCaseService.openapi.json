{"components":{"schemas":{"HeaderRequest":{"description":"Simple request message","properties":{"data":{"description":"Request data","type":"string"}},"type":"object"},"HeaderResponse":{"description":"Simple response message","properties":{"result":{"description":"Response data","type":"string"}},"type":"object"}}},"info":{"title":"EdgeCaseService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/edge-headers/complex":{"post":{"description":"Method with complex header combinations","operationId":"ComplexHeaders","parameters":[{"description":"Array header with complex format","in":"header","name":"X-Complex-Array","required":true,"schema":{"type":"array"}},{"description":"Edge case header","in":"header","name":"X-Edge-Case","required":false,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ComplexHeaders","tags":["EdgeCaseService"]}}}}